### PR TITLE
refactor: tidy up the code

### PR DESCRIPTION
**Pull request by Simon-He95**

### CODE REVIEW

File: src/createComposable.ts [hunk 1]
```diff
@@ -254,11 +253,6 @@ function createWatcher(
...
   const segments = path.split('.')
-  return () => {
-    let cur = ctx
-    for (let i = 0; i < segments.length && cur; i++) {
-      cur = cur[segments[i]]
-    }
-    return cur
-  }
+  return () =>
+    ctx ? segments.reduce((cur, item) => (cur = cur[item]), ctx) : ctx
```
Comment (LinusBorg) [inline src/createComposable.ts]: KI'd prefer to keep the for loop. it's more expressive in general faster than reduce.

File: src/inject.ts [hunk 2]
```diff
@@ -68,12 +68,10 @@ export function resolveInjections(
...
-  if (isArray(raw)) {
-    const res: ObjectInjectOptions = {}
-    for (let i = 0; i < raw.length; i++) {
-      res[raw[i]] = raw[i]
-    }
-    return res
-  }
-  return raw
+  return isArray(raw)
+    ? raw.reduce((res, item) => {
```
Comment (LinusBorg) [inline src/inject.ts]: I'd prefer to keep the for loop. it's more expressive in general faster than reduce.

**Review by LinusBorg — CHANGES_REQUESTED** [review]
Thanks for this cleanup! If you can revert the two changes about the for loops, I'm happy to merge it.

File: src/createComposable.ts (hunk 1 above)
Comment (Simon-He95) [reply src/createComposable.ts]: of course

File: src/inject.ts (hunk 2 above)
Comment (Simon-He95) [reply src/inject.ts]: I've switched back to for, but with a little change, I think it's better to minimize the use of for in the if condition

**Review by LinusBorg — APPROVED** [review]

**Pull request merged**